src/components/Select/Select.tsx

```diff
@@ -28,51 +27,64 @@ export const Select: FC<SelectProps> = ({
 }) => {
     const [selected, setSelected] = useState<string>(initialValue);
     const [query, setQuery] = useState("");
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const comboboxButtonRef = useRef<HTMLButtonElement | null>(null); // Added ref for Combobox.Button
+
+    useEffect(() => {
+        if (onChange) onChange(selected);
+    }, [selected, onChange]);
```

Comment:
Why this?

Reply:
![image](https://github.com/LaurierHawkHacks/Dashboard/assets/46619361/41766465-bab5-4f31-b933-566a859e8886)

The useEffect is causing this issue

Reply:
The onChange prop passed from pages set another state too so this complicates things. If we make it that each component controlled their own values, we can get away with using a ref to store the application data.